src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -804,8 +804,21 @@ protected bool TakeStartLine(SocketInput input)
                 QueryString = queryString;
                 HttpVersion = httpVersion;
 
-                bool caseMatches;
+                if (requestUrlPath.Length > 0 && requestUrlPath[0] != '/')
+                {
+                    Uri uri;
+                    if (Uri.TryCreate(requestUrlPath, UriKind.Absolute, out uri))
```

Comment:
Not optimal but I don't see the need to optimize for this case - requests for absolute URIs are not expected to be the norm. So most of the time we won't be running this.